src/js/extension.js

```diff
@@ -66,8 +66,124 @@ var Extension;
             // has a .base value pointing to the editor
             // owning us. has been given a .name if no
             // name present
-        }
-        // core API definition for an "extension" follows:
+        },
+
+        /* parent: [boolean]
+         *
+         * setting this to true will set the .base property
+         * of the extension to be a reference to the
+         * medium-editor instance that is using the extension
+         */
+        parent: false,
+
+        /* base: [MediumEditor instance]
+         *
+         * If .parent is set to true, this will be set to the
+         * current MediumEditor instance before init() is called
+         */
+        base: null,
+
+        /* name: [string]
+         *
+         * 'name' of the extension, used for retrieving the extension.
+         * If not set, MediumEditor will set this to be the key
+         * used when passing the extension into MediumEditor via the
+         * 'extensions' option
+         */
+        name: null,
+
+        /* checkState: [function (node)]
+         *
+         * If implemented, this function will be called one or more times
+         * the state of the editor & toolbar are updated.
+         * When the state is updated, the editor does the following:
+         *
+         * 1) Find the parent node containing the current selection
+         * 2) Call checkState on the extension, passing the node as an argument
+         * 3) Get tha parent node of the previous node
+         * 4) Repeat steps #2 and #3 until we move outside the parent contenteditable
+         */
+        checkState: null,
+
+        /* getButton: [function ()]
+         *
+         * If implemented, this function will be called when
+         * the toolbar is being created.  The DOM Element returned
+         * by this function will be appended to the toolbar along
+         * with any other buttons.
+         */
+        getButton: null,
```

Comment:
I've been trying to think about this a bit, specifically because of extensions that support 'forms' as well. A Button extension would extend Extension.  But should a FormExtension extend Button or extend Extension?  I'm thinking it might be better for FormExtension to extend from Extension since it's probably unnecessary for it to share all the same code as the Button.  In that case, it would be good to keep `getButton` defined here.

I think the value in Button is that it's a really simple way to create a button that just maps to a `document.execCommand` command.  That's what I originally called it `DefaultButton` instead of button because it doesn't really help you out a lot if you're trying to create a button that does something other than just call `execCommand`.

I also look at an example like [this](https://github.com/daviferreira/medium-editor/wiki/Custom-Buttons-and-Extensions#highligher-extension-using-checkstate) and I think it might be ok to leave `getButton` in the base extension, because of the simplicity for some cases...

Because of this, I'm wondering if we should even have a Button extension right now. Just some thoughts, I might just be rambling and not giving a good argument. 

Either way, if we create a Button extension, that'll happen in another pull request, and if we want to remove `getButton` from Extension then, we still can.